app/repository/RestaurantRepository.java

```diff
@@ -2,10 +2,39 @@
 
 import com.google.inject.ImplementedBy;
 import models.Restaurant;
+import models.RestaurantTable;
+import org.hibernate.Criteria;
+import org.hibernate.Session;
+import org.hibernate.criterion.Restrictions;
+import org.hibernate.jpa.HibernateEntityManager;
+import play.db.jpa.JPA;
+
+import java.util.List;
 
 /**
  * Created by lapa on 5/6/17.
  */
-@ImplementedBy(RestaurantRepositoryImplementation.class)
-public interface RestaurantRepository extends Repository<Restaurant> {
+//@ImplementedBy(RestaurantRepositoryImplementation.class)
+public class RestaurantRepository extends RepositoryImplementation<Restaurant> {
+
+        public static Criteria getCriteria(){
```

Comment:
This should also belong to `RepositoryImplementation` class that you extend, so you don't have to implement this in each repository class.